File: resources/src/Matroos.Resources/Utilities/DataType.cs
using Matroos.Resources.Attributes;

namespace Matroos.Resources.Utilities;

public enum DataType
{
    [AType(typeof(bool))]
    BOOLEAN,    // 0
    [AType(typeof(string))]
    DATE,       // 1
    [AType(typeof(double))]
    DOUBLE,     // 2
    [AType(typeof(int))]
    INTEGER,    // 3
    [AType(typeof(string))]
    STRING,     // 4
    [AType(typeof(List<object>))]
    LIST        // 5
}

File: frontend/src/utils/dataType.ts
import { DataType } from './types';

export function DataTypeToString(type: DataType): string {
    switch (type) {
        case DataType.BOOLEAN:
            return 'Boolean';
        case DataType.DATE:
            return 'Date';
        case DataType.DOUBLE:
            return 'Double';
        case DataType.INTEGER:
            return 'Integer';
        case DataType.STRING:
            return 'String';
        case DataType.LIST:
            return 'List';
    }
}
